Add opacity setting to text / node circle of falsey values

njhoffman opened the pull request:
Added a couple of styling options to apply opacity values to null/empty/undefined values.  Defaults are 1.0 opacity for all nodes which can be changed with supplying property to the chart component as with other styling modifications.

![image](https://user-images.githubusercontent.com/3154616/40633243-faefc324-62b3-11e8-9c5b-dc53a0903c45.png)

njhoffman committed:
Add option to apply opacity to node and text elements that have a falsey value

romseguy commented on src/charts/tree/tree.js:
```diff
@@ -16,14 +16,22 @@ const defaultOptions = {
+      },
+      opacity: {
+        'default': 1.0,
+        empty: 1.0
```
shouldn't opacity be lower than 1.0 for empty nodes?

njhoffman replied:
Sorry about the late reply, I didn't want to assume people would want empty nodes faded, so I kept the default option as the same.  I pushed another commit with a fix and have the empty nodes and text set at 0.75 so it is not too noticeable but enough for the feature to be noticed (I find it really helpful to keep it below 0.5 so I can see nodes of interest).  I will post more details to the pull request.

njhoffman committed:
Fix nodes misidentified as empty when they contain an object in the "object" property, minor refactoring

njhoffman committed:
Update README.md to correct and add to outdated style options info

njhoffman commented:
I noticed a bug with my changes, objects are stored in a different property (node.object) than the primitives (node.value). This should fix that, I also updated the readme to include up to date information..

Bug: 
![image](https://user-images.githubusercontent.com/3154616/42236147-aa75105c-7ebf-11e8-8f61-019b96ba508a.png)

Fixed:

![image](https://user-images.githubusercontent.com/3154616/42236267-0b92776c-7ec0-11e8-823d-674fe8d089bd.png)

 I have falsey nodes/texts set at opacity 0.75 so it's not too noticeable for those who may not care for it.  Here's what 0.75 looks like for me (personally I like it below 0.5 to clearly highlight values of interest).

![image](https://user-images.githubusercontent.com/3154616/42236338-3e79b230-7ec0-11e8-9505-a05bfd8d7344.png)

njhoffman commented:
I have a few more enhancements to the graph I will be making pull requests for soon.

romseguy commented:
Superseded by https://github.com/reduxjs/d3-state-visualizer/pull/17
